src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1217,6 +1218,41 @@ public override int GetHashCode()
             return (this.Name.GetHashCode());
         }
 
+        internal static unsafe int GetIgnoreCaseHash(string source)
+        {
+            // Do not allocate on the stack if string is empty
+            if (source.Length == 0)
+            {
+                return source.GetHashCode();
+            }
+
+            Span<char> span = source.Length < 256 ? 
+                stackalloc char[source.Length] : 
+                ArrayPool<char>.Shared.Rent(source.Length);
+
+            if (!source.IsAscii())
+            {
+                TextInfo.Invariant.ToLower(source).AsSpan().CopyTo(span);
```

Comment:
This should use the newly added no-allocation Span-based API. No need to allocated an extra memory for the string.